File: LeetCode/neetcode-150/array-hashing/two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        ind_dict = {}
        for i, num in enumerate(nums):
            ind_dict[num] = i

        for i, num in enumerate(nums):
            j = ind_dict.get(target - num, -1)
            if j >= 0 and j != i:
                return [i, j]
            
        

File: LeetCode/neetcode-150/array-hashing/two-sum/two-sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    for (int i = 0; i < numsSize; i++){
        for (int j = i + 1; j < numsSize; j++){
            if (nums[i] + nums[j] == target){
                int * ret = (int *) malloc(2 * sizeof(int));
                ret[0] = i;
                ret[1] = j;
                *returnSize = 2;
                return ret;
            }
        }
    }
    return NULL;
}
